Description: Fix OOB access on some UTF input
 On architectures with signed char type and input that is >=128 there is
 an out-of-bounds access causing SIGSEGV. It is most probably not exploitable
 but degrades user experience.
---
Origin: upstream, https://github.com/Tomas-M/iotop/commit/8aaa4fce743cf14a5a727c6cb24c63450d317a28
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/iotop/+bug/1932523
Forwarded: not-needed
Last-Update: 2021-09-06

--- iotop-c-1.17.orig/src/utils.c
+++ iotop-c-1.17/src/utils.c
@@ -171,7 +171,7 @@ inline const char *esc_low_ascii1(char c
 	static char ehex[0x20][6];
 	static int initialized=0;
 
-	if (c>=0x20) // no escaping needed
+	if (c<0||c>=0x20) // no escaping needed
 		return NULL;
 	if (!initialized) {
 		int i;
